src/auth/mock-auth.guard.ts

```diff
@@ -5,14 +5,15 @@
  */
 
 import { ExecutionContext, Injectable } from '@nestjs/common';
+import { UsersService } from '../users/users.service';
 
 @Injectable()
 export class MockAuthGuard {
-  canActivate(context: ExecutionContext) {
+  constructor(private usersService: UsersService) {}
+
+  async canActivate(context: ExecutionContext) {
     const req = context.switchToHttp().getRequest();
-    req.user = {
-      userName: 'hardcoded',
-    };
+    req.user = await this.usersService.createUser('hardcoded', 'Testy');
```

Comment:
This might lead to the user getting created multiple times.
It should be safer to move the `createUser` call to the constructor.

Reply:
good point. I'll change that

Reply:
Only problem is: This is a async code that can't be called in the consturctor. The normal solution would be to add a async builderPattern, but as we don't instantiate the class our self this won't do as any good. Only solution I could think of would be a singleton, not the most elegant solution, but I guess it should do.

Reply:
You could check if the user already exists?

Reply:
Would a boolean field enough for this check?

Reply:
If I understand the tests correctly, the db gets recreated after each test so this way should suffice.

Reply:
Is this resolved?

Reply:
For me yes. But I'd leave the closing of this conversation to @davidmehren